tests/Rules/AllowedCharactersUsernameTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+use ARKEcosystem\Fortify\Rules\AllowedCharactersUsername;
+
+it('can pass validation', function () {
+    $subject = new AllowedCharactersUsername();
+
+    expect($subject->passes('name', 'foo'))->toBeTrue();
+    expect($subject->passes('name', 'foo_bar'))->toBeTrue();
+    expect($subject->passes('name', 'foo_bar='))->toBeTrue();
+});
+
+it('can fail validation', function () {
+    $subject = new AllowedCharactersUsername();
+
+    expect($subject->passes('name', '.'))->toBeFalse();
+    expect($subject->passes('name', '..'))->toBeFalse();
+    expect($subject->passes('name', '=!.'))->toBeFalse();
```

Comment:
this will need some more testcases to ensure it all works as intended, e.g. `foo__bar`, `foo_bar_____`, `a_______`, `a-----------a`, etc

Reply:
@ItsANameToo Will do once we settled what we want :vinigotit: